### DISCUSSION TITLE
Remote config conditional does not work properly with audiences

### DISCUSSION THREAD

**Discussion by vshjxyz** [invertase/react-native-firebase#4795] [Q&A]
Hello, I've been using this great library for a while. Now I need to do a remote configuration only for people that emitted an event with a specific value. I've managed to create a new audience to filter out people with that event + value but I'm facing a large number of issues, been debugging this the past 2 days and so far:

- I intermittently get either the default config value or the audience-specific value
  - e.g. value MY_CUSTOM_CONFIG with value `0` default and `1` if my user is in the desired audience
      - I get 0 most of the times, even if I change user via `analytics().setUserId` multiple times
      - sometimes it switches to 1 (the desired behavior) but it STAYS to 1 even if I change user id via analytics
- Audiences are not reflecting the correct number of users, I've fired the event multiple times with at least 5+ users, although in the audience (after 24h) I can only see 2 users matching it.
- Trying with those 2 users does not change the issue, I almost always get `0` instead of `1`

I'm in debug mode (can see the event in analytics debug view and firebase debug view) AND I have specifically turned on `setAnalyticsCollectionEnabled` as I've read it won't collect audience data otherwise

I'm running out of options and starting to convert this to a user property (without audiences) **which behaves as I expected and it switches immediately when I switch users** - does anyone got any experience / idea on how is it meant to work?

**Comment by mikehardy** [comment]
I have also found audiences very difficult to work with. And remote config delays / slowness compounds it. I can only recommend watching all the firebase developer videos on the subject. They are illuminating. I use firebase properties personally

**Reply by vshjxyz** [reply]
Thanks for your quick reply! do you mean user properties? I've noticed that they are actually working well w/ remote config conditions, but the issue there is that there Is no persistence so I need to re-set them all the times! 
I'd really need the audiences to work as expected. So far I've seen the configs are fetched with default values the first time I pull the application. 

After around 1 minute trying to poll new values, **I finally get the expected ones if I've set the user id through analytics with the id of a user that matches an audience** - This is fine to me but it also seems that logging out and logging in w another user (different call to analytics w another user ID), the config values are still there, persisted **until I remove the application and install it back**. It seems that remote config is bound to the app installation, while audiences are bound to the analytics user ID 🤔 

https://support.google.com/firebase/answer/6318039?hl=en

**UPDATE:**
the rnfirebase iid package has a `remove` method that I'm able to call in order to force a regeneration of the ID when logging out (thus removing the audience concept), I think this is what I missed in the flow and I'm now able to logout/login and get the expected values from remote config within the audience of the user (it takes always some minutes to get the right values tho, but should be fine for my use cases)

looking forward to hearing if other people solved it in another way or how can we get the right values earlier

**Reply by mikehardy** [reply]
I mean that I have an app-specific 'User' object in firestore, and I store information related to the user there, then I use cloud functions to select groups of users and do something. Which is to say, I give up nearly all the power of remote config and audiences and ability to delegate marketing things to an actual marketing group, which is sub-optimal of course but also gets around the issues ;-)

Your solution sounds interesting - I'm only worried that the iid package itself is deprecated. Have you checked the new upstream Installations module to see if it has a similar method? If it does - no problem - since that is the future of the "iid" APIs, if not then be careful investing in the solution as the functionality will eventually be removed owing to upstream deprecation

I am also interested in hearing others experiences though I fear there may not be many, the DIscussions section is mostly "lots of questions" and "me answering them as best as possible" ;-).

Did you find anything illuminating in the firebase youtube channel? Those videos really are very well done in my experience

**Reply by vshjxyz** [reply]
Awesome, good point on the iid deprecation, I've checked and it seems like https://firebase.google.com/docs/reference/android/com/google/firebase/installations/FirebaseInstallations this will substitute the API and contains the same `delete` method (maybe forcing a refreshToken works as well, although I don't use firebase auth so delete might be more safer) so I think I should be kinda fine as long as I can deal with audience / remote config latency and rely on them being "eventually right". It'll be hard to test but will give it a go! 

I agree the youtube channel is very informative, although I think firebase pushes more for the full integration (I expect this would work much better if I used firebase auth and users) so it's hard to find information for an approach that ONLY uses analytics + remote configs + audiences 🤔  I'll try to keep this updated if I have news
